packages/core/src/storage/migration/__tests__/0.2.test.ts

```diff
@@ -0,0 +1,144 @@
+import type { FileSystem } from '../../..'
+
+import { unlinkSync, readFileSync } from 'fs'
+import path from 'path'
+
+import { InMemoryStorageService } from '../../../../../../tests/InMemoryStorageService'
+import { Agent } from '../../../../src'
+import { agentDependencies } from '../../../../tests/helpers'
+import { InjectionSymbols } from '../../../constants'
+import { DependencyManager } from '../../../plugins'
+import * as uuid from '../../../utils/uuid'
+import { UpdateAssistant } from '../UpdateAssistant'
+
+const backupDate = new Date('2022-01-21T22:50:20.522Z')
+jest.useFakeTimers().setSystemTime(backupDate)
+const backupIdentifier = backupDate.getTime()
+
+const walletConfig = {
+  id: `Wallet: 0.2 Update`,
+  key: `Key: 0.2 Update`,
+}
+
+describe('UpdateAssistant | v0.2 - v0.3', () => {
+  it(`should correctly update proof records and create didcomm records`, async () => {
+    // We need to mock the uuid generation to make sure we generate consistent uuids for the new records created.
+    let uuidCounter = 1
+    const uuidSpy = jest.spyOn(uuid, 'uuid').mockImplementation(() => `${uuidCounter++}-4e4f-41d9-94c4-f49351b811f1`)
+
+    const aliceCredentialRecordsString = readFileSync(
+      path.join(__dirname, '__fixtures__/alice-4-proofs-0.2.json'),
+      'utf8'
+    )
+
+    const dependencyManager = new DependencyManager()
+    const storageService = new InMemoryStorageService()
+    dependencyManager.registerInstance(InjectionSymbols.StorageService, storageService)
+
+    const agent = new Agent(
+      {
+        config: {
+          label: 'Test Agent',
+          walletConfig,
+        },
+        dependencies: agentDependencies,
+      },
+      dependencyManager
+    )
+
+    const fileSystem = agent.injectionContainer.resolve<FileSystem>(InjectionSymbols.FileSystem)
+
+    const updateAssistant = new UpdateAssistant(agent, {
+      v0_1ToV0_2: {
+        mediationRoleUpdateStrategy: 'doNotChange',
+      },
+    })
+
+    await updateAssistant.initialize()
+
+    // Set storage after initialization. This mimics as if this wallet
+    // is opened as an existing wallet instead of a new wallet
+    storageService.records = JSON.parse(aliceCredentialRecordsString)
+
+    expect(await updateAssistant.isUpToDate()).toBe(false)
+    expect(await updateAssistant.getNeededUpdates()).toEqual([
+      {
+        fromVersion: '0.2',
+        toVersion: '0.3',
+        doUpdate: expect.any(Function),
+      },
+    ])
+
+    await updateAssistant.update()
+
+    expect(await updateAssistant.isUpToDate()).toBe(true)
+    expect(await updateAssistant.getNeededUpdates()).toEqual([])
+
+    // MEDIATOR_ROUTING_RECORD recipientKeys will be different every time, and is not what we're testing here
+    delete storageService.records.MEDIATOR_ROUTING_RECORD
```

Comment:
We've recently had this issue in #985 and I've ended up mocking IndyWallet's createDid method to create a consistent routing record. 

Even if my preference would be to keep the database as loaded from file, your solution is simpler and probably better considering a future decoupling from IndyWallet, so maybe would be nice to implement it in 0.1.test.ts as well.